src/librustc_lint/types.rs

```diff
@@ -267,7 +267,19 @@ fn lint_int_literal<'a, 'tcx>(
         }
 
         cx.struct_span_lint(OVERFLOWING_LITERALS, e.span, |lint| {
-            lint.build(&format!("literal out of range for `{}`", t.name_str())).emit()
+            lint.build(&format!("literal out of range for `{}`", t.name_str()))
+                .note(&format!(
+                    "the literal `{}` does not fit into type `{}` whose range is {}..={}",
```

Comment:
Since the range is a valid rust expression, I would write in backticks.

```suggestion
                    "the literal `{}` does not fit into the type `{}` whose range is `{}..={}`",
```